nca/Utils/ExplTracker.py

```diff
@@ -0,0 +1,419 @@
+        # also use one peer for each deployment
+        peer_names = set()
+        for peer in self.all_peers:
+            peer_names.add(peer.full_name())
+
+        for peer1 in peer_names:
+            for peer2 in peer_names:
+                if peer1 == peer2:
+                    text = self.explain([peer1])
+                else:
+                    text = self.explain([peer1, peer2])
+                # Create the XML entry element
+                entry = soup.new_tag('entry')
+                entry_id += 1
+                entry['id'] = str(entry_id)
+                entry['src'] = self.get_printout_ep_name(peer1)
+                entry['dst'] = self.get_printout_ep_name(peer2)
+                text_elem = Tag(soup, name='text')
+                text_elem.string = text
+                entry.append(text_elem)
+                soup.append(entry)
+        return soup.prettify()
+
+    def get_working_ep_name(self, name):
+        """
+        if ep is in 'deployments' mode, the given name will be the workload name but the full name is always used as index
+        :param name: str: the name to convert to full_name (it it is not already the full name)
+        :return: str: full name of the element
+        """
+
+        # Replace '[' with '(' and ']' with ')'
+        name = name.replace('[', '(').replace(']', ')')
+
+        if self.ep == 'deployments':
+            # convert from workload name to fullname
+            for fullname, data in self.ExplDescriptorContainer.items():
+                workload_name = data.get('workload_name')
+                if name == workload_name or name == fullname:
+                    # found the workload name, return its fullname
+                    # or, it has no workload name
+                    return fullname
+            return ''
+        else:
+            # we are in 'pods' mode so the name is already the fullname
+            return name
+
+    def explain(self, nodes):
+        """
+        The magic function to explain the connectivity or the LACK of it between the given nodes
+        It has 2 modes:
+            single node - if a single node is given, all the configurations on that node are displayed.
+            two nodes - if 2 nodes are given, either they hava a connection between them and the configurations responsible for
+                        the connection are displayed. or, they lack a connection, in which case, all affecting configurations
+                        on those 2 nodes are displayed.
+        :param list(str) nodes: nodes to explain
+        :return: str: the explanation out string
+        """
+        if len(nodes) < 1:
+            return ''
+        elif len(nodes) > 2:
+            NcaLogger().log_message(f'Explainability error: only 1 or 2 nodes are allowed for explainability query,'
+                                    f' found {len(nodes)} ', level='E')
+            return ''
+
+        if nodes[0] == 'ALL':
```

Comment:
The third mode? Should be added to the description

Reply:
Done (added the 'ALL' option to the description)